packages/lib/modules/swap/SwapForm.tsx

```diff
@@ -94,6 +102,22 @@ export function SwapForm() {
     }
   }
 
+  function handleTokenSelectForPoolSwap(token: GqlToken) {
+    if (
+      tokens.length === 2 &&
+      tokenSelectKey === 'tokenIn' &&
+      isSameAddress(token.address, tokenOut.address)
+    )
+      return switchTokens()
+    if (
+      tokens.length === 2 &&
+      tokenSelectKey === 'tokenOut' &&
+      isSameAddress(token.address, tokenIn.address)
+    )
```

Comment:
Shouldn't this be handled in the same we were handling it in the swap UI? i.e. you just can't select the same token that's currently selected?

Reply:
Seems like it should be logic that's handled in the modal, not here.

Reply:
I wanted to reuse the same selector that we are using to choose between native asset and wrapped native asset where you can chose the currently selected token.

The logic is handling the case when you chose the other token in a 2-token pool swap. So instead of disabling the other token, it switches them:

https://github.com/user-attachments/assets/259e5c66-1c9a-4abc-a393-975947dc2ea1